<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Novartis Quantiitative Sciences Hackathon : Hack 2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="mailto:&lt;as5697@columbia.edu&gt;">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/apoorvasrinivasan26/hack_2">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Novartis Quantiitative Sciences Hackathon : Hack 2</h1>

</div>


<div class="figure">
<img src="franki-chamaki-1K6IQsQbizI-unsplash.jpg" />

</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>In this project presented by Novartis during their 2019 Academia to Industry Quantitative Sciences Hackathon, I</p>
<ul>
<li><p>Compare Support Vector Machine with Recursive Feature Elemination(SVM-RFE) with a modified version of SVM-RFE with t-statistic(SVM-T-Statistic) to identify and classify proteins from SOMAscan that are most predictive of weight loss in obese monkeys.</p></li>
<li><p>Achieved higher classification accuracy with SVM-T-RFE method at dose 0.1mg/ml</p></li>
<li><p>Identified top proteins for classification using both methods</p></li>
</ul>
</div>
<div id="problem" class="section level2">
<h2>Problem</h2>
<p>Can we find the identify proteins that are most predictive of weight loss from SOMAscan assay in obese monkeys at a certain dose?</p>
</div>
<div id="data-description-and-pre-processing" class="section level2">
<h2>Data description and pre-processing</h2>
<p>The data we were presented were protein abundance measurements(rfu2) of 5080 proteins from SOMAscan assay generated simultaneously. These measurements were from 33 animals from 3 different studies injected with 8 different doses at 4 timepoints(day 0,1 ,4 and 7). Compound blood concentration at these different timepoints was also made available. We note that some of these monkeys were reused for these different studies.</p>
<p>For the purpose of our analysis, we assume that the animals reused for different studies were independent since they were one year apart. Since, we’re interested in looking at dose-response relationship, our first task was to identify an appropriate dose to conduct our analysis.</p>
<p>The idea was to pick a dose where we has sufficient sample size and where log pk drug concentration wasn’t to an extent where it could result in over-expression of protein. From Figure 1, we can see that dose 0.1 has the maximum number of animals(12) so we have sufficient sample size and it results in reasonable drug concentration in blood.</p>
<p><strong>percentage body weight change was recoded into factors where they were categorized as significant body weight loss(&lt; -1) and little or no body weight loss(&gt;= -1) in order to perform below described classification problem.</strong> All the rfu2 values were log transformed. Protein measurements at timepoint 0 was excluded because we wanted to study the effects after the dose had been administered.</p>
<div class="figure">
<img src="dose_conc_frames.png" alt="Figure 1" width="500" />
<p class="caption">Figure 1</p>
</div>
</div>
<div id="approach" class="section level2">
<h2>Approach</h2>
<p>Since we have small sample and thousands of protein abundance data, the selection of those proteins that are important for distinguishing the different sample classes being compared, poses a challenging problem in high dimensional data analysis.</p>
<p>Below are some properties I based my model selection on:</p>
<ul>
<li><p>Scale to very high dimensions: methods should continue to generalize well even if the number of observed factors p significantly exceeds the number of observations n (p ≫ n)</p></li>
<li><p>Be interpretable: the output produced by the methods should be directly viable of biological and clinical interpretation, so black-box methods or complex manipulations of the features wasn’t done</p></li>
<li><p>􏰀Biologically, protein expression is done in cascades. Thus, we would expect groups of proteins to be highly correlated with each other, representing the interplay of up and down-regulation. Hence, it is important to analyze the data jointly: low-dimensional probabilistic models or statistical tests ignore complex relationships between the predictors. We can see this from the correlation plot of 500 random proteins below.</p></li>
</ul>
<div class="figure">
<img src="Screen%20Shot%202019-09-10%20at%204.08.06%20PM.png" alt="Figure 2" />
<p class="caption">Figure 2</p>
</div>
<p>SVM-RFE[1] is recognized as one of the most effective filtering methods for genes in the bioinfomatics world. It is based on an algorithm that finds the best possible combination for classification. The drawback of it, however, is that it does it without considering the differentially significant features between the classes. Features with notable classification accuracy and also differentially significant between the classes have a predominant role in a biological aspect. Hence, to overcome this limitation, in the proposed algorithm[2], t-statistics is included along with SVM-RFE.</p>
</div>
<div id="svm-trfe-algorithm" class="section level2">
<h2>SVM-TRFE algorithm</h2>
<ol style="list-style-type: decimal">
<li><p>The coefficient and the expression mean difference between two compared groups of each individual feature is calculated using welch’s t-test which is calculated as <img src="t_stat.png" /></p></li>
<li><p>It is incorporated into the weight value of that particular gene or the feature by <img src="Screen%20Shot%202019-09-11%20at%206.37.08%20PM.png" /> where <span class="math inline">\(t_i\)</span> is the t-statistics between the two comapred groups and <span class="math inline">\(w_i\)</span> is the SVM-RFE weight vector defined by Guyon et el. [7] and θ is the parameter determining the tradeoff between SVM weight score and T-statistic, and θ is the value between 0 and 1.</p></li>
<li>The iteration process is followed by backward removal of the feature.</li>
<li><p>The iteration process is continued until there is only one feature remaining in the dataset. 4 )The smallest ranking weight will be removed by the algorithm and stored in a stack, while remaining the feature variables have a significant impact for producing weight. 5 )Finally, the feature variables which are stored in the stack will be listed in the descending order of descriptive different degree.</p></li>
</ol>
</div>
<div id="modelling" class="section level2">
<h2>Modelling</h2>
<p>In order to compare the performance of both SVM-T-RFE and SVM-RFE in classification, a histogram of p values of the selected top 100 proteins is plotted. The p values in the left histogram were calculated using t-statistic. We can see that for the model with t-statistics, all the p-values for the top 100 proteins is less than the generally accepted 0.05 while the range is much larger for the model with SVM-RFE. This is further illustrated by the boxplot below with p-value of 0.3 as the median for the second model. (done using sigFeature package in R)</p>
<div class="figure">
<img src="Pvalue_hist_dose0.1.png" alt="Figure 3" />
<p class="caption">Figure 3</p>
</div>
<div class="figure">
<img src="Box%20plot_pvalue_d0.1.png" alt="Figure 4" />
<p class="caption">Figure 4</p>
</div>
</div>
<div id="validation" class="section level2">
<h2>Validation</h2>
<p>More work needs to be done in this regard since validating CV error using k-fold is computationally expensive.</p>
<p>However, to get a sense of model performance, the data was split into 80% training and 20% testing and the parameters from Table 1 were calculated with top 1000 proteins from each of these models.</p>
<div class="figure">
<img src="svm-t-rfe.png" alt="Table 1" />
<p class="caption">Table 1</p>
</div>
<p>We can see that SVM-RFE with t-statistics does a far better job of classifying weight loss than just SVM-RFE.</p>
<div id="list-of-top-20-proteins" class="section level3">
<h3>List of Top 20 Proteins</h3>
<p>Below are the list of top proteins from SVM-RFE and SVM-T-RFE. Interestingly, we notice that the target 1620 shows up as a top protein from the SVM-RFE model. This needs to be further investigated in this future.</p>
<div class="figure">
<img src="svm_rfe_highligh_rank.png" alt="Top 20 Proteins from SVM-RFE" />
<p class="caption">Top 20 Proteins from SVM-RFE</p>
</div>
<div class="figure">
<img src="svm_t_pval.png" alt="Top 20 Proteins from SVM-T-RFE" />
<p class="caption">Top 20 Proteins from SVM-T-RFE</p>
</div>
</div>
</div>
<div id="conclusion-and-future-directions" class="section level2">
<h2>Conclusion and Future Directions:</h2>
<p>High-demsional reduction is biomarker discovery is a hard problem. SVM-RFE and it’s modified method SVM-T-RFE seem promising to identify suitable biomakers with reasonable predictive accuracy. Since project was done to satisfy a curiosity to test out an approach usually used in gene expression data with SOMAscan protein expressions, I make no claims about the robustness of the identified proteins in playing a role in weight loss. However, it has brought certain proteins and dose to attention which can be further studied.</p>
<p>For the future steps,</p>
<ul>
<li><p>Validate this approach from different datasets of SOMAscan protein expression</p></li>
<li><p>Combine data-driven and expert knowledge-based approaches to concurrently mine the power of statistical algorithms for selection while guiding the process to include weighting factors derived from expert-knowledge. One way is to use bayesian approaches since they have always included weighting factors by assigning prior probabilities.</p></li>
<li><p>Explore different modifications of SVM-RFE such as SVM-RCE(support vector machine with recursive cluster elimination)[5] or SVM-RFE OA[6] since Support vector machine-recursive feature elimination (SVM-RFE) is an efficient feature selection technique that has shown its power in many applications.</p></li>
</ul>
</div>
<div id="references" class="section level2">
<h2>References:</h2>
<ol style="list-style-type: decimal">
<li>Kai-Bo Duan, J. C. Rajapakse, Haiying Wang and F. Azuaje, “Multiple SVM-RFE for gene selection in cancer classification with expression data,” in IEEE Transactions on NanoBioscience, vol. 4, no. 3, pp. 228-234, Sept. 2005.</li>
<li>Das, P., Roychoudhury, S., &amp; Tripathy, S. (2017). sigFeature: An R-package for significant feature selection using SVM-RFE &amp; t-statistic. Canadian Journal of Biotechnology, 1, 35. <a href="doi:http://dx.doi.org/10.24870/cjb.2017-a22" class="uri">doi:http://dx.doi.org/10.24870/cjb.2017-a22</a></li>
<li>Zheng Rong Yang, Biological applications of support vector machines, Briefings in Bioinformatics, Volume 5, Issue 4, December 2004, Pages 328–338, <a href="https://doi.org/10.1093/bib/5.4.328" class="uri">https://doi.org/10.1093/bib/5.4.328</a></li>
<li>Xiaobo Li, Sihua Peng, Jian Chen, Bingjian Lü, Honghe Zhang, Maode Lai, SVM–T-RFE: A novel gene selection algorithm for identifying metastasis-related genes in colorectal cancer using gene expression profiles, Biochemical and Biophysical Research Communications</li>
<li>Yousef M, Jung S, Showe LC, Showe MK. Recursive cluster elimination (RCE) for classification and feature selection from gene expression data. BMC Bioinformatics. 2007;8:144. Published 2007 May 2. <a href="doi:10.1186/1471-2105-8-144" class="uri">doi:10.1186/1471-2105-8-144</a></li>
<li><a href="https://www.mdpi.com/248888" class="uri">https://www.mdpi.com/248888</a></li>
<li>I. Guyon, J. Weston, S. Barnhill, V. Vapnik Gene selection for cancer classification using support vector machines Machine Learning, 46 (2002), pp. 389-422</li>
</ol>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
